hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java

```diff
@@ -1002,7 +1002,7 @@ private RegionLocations locateRegionInMeta(TableName tableName, byte[] row, bool
       }
       // Query the meta region
       long pauseBase = connectionConfig.getPauseMillis();
-      takeUserRegionLock();
+      final long lockStartTime = takeUserRegionLock();
```

Comment:
I know I suggested this as a possible approach in the other comment, but it also looks weird to me now 🙃 

I'd be okay with it as-is, but it may also just make sense to leave the return type as `void` and keep all the context within this method, as in:
```suggestion
      takeUserRegionLock();
      final long lockStartTime = EnvironmentEdgeManager.currentTime();
```

Reply:
Yes, let's keep the original type `void` for the function.